Using the REPL to view the AST results in a `ParserException`

## Description
One of the [REPL features](https://github.com/partiql/partiql-lang-kotlin/wiki/Command-Line-Tutorial#advanced-repl-features) is to view the AST of a PartiQL statement using `!!`. Currently, it results in a parser exception:

```
Welcome to the PartiQL REPL!
Typing mode: LEGACY
Using version: 0.8.2-SNAPSHOT-6550a75d
PartiQL> 1 + 1
   | !!
org.partiql.lang.syntax.ParserException: mismatched input '!' expecting {<EOF>, ';'}
	Parser Error: at line 2, column 1: unexpected token found, IDENTIFIER : '!'

	at org.partiql.lang.syntax.PartiQLParser$ParseErrorListener.syntaxError(PartiQLParser.kt:190)
	at org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41)
	at org.antlr.v4.runtime.Parser.notifyErrorListeners(Parser.java:543)
	at org.antlr.v4.runtime.DefaultErrorStrategy.reportInputMismatch(DefaultErrorStrategy.java:327)
	at org.antlr.v4.runtime.DefaultErrorStrategy.reportError(DefaultErrorStrategy.java:139)
	at org.partiql.lang.syntax.antlr.PartiQLParser.statement(PartiQLParser.java:630)
	at org.partiql.lang.syntax.antlr.PartiQLParser.root(PartiQLParser.java:367)
	at org.partiql.lang.syntax.PartiQLParser.parseQuery$lang(PartiQLParser.kt:100)
	at org.partiql.lang.syntax.PartiQLParser.parseQuery(PartiQLParser.kt:90)
	at org.partiql.lang.syntax.PartiQLParser.parseAstStatement(PartiQLParser.kt:61)
	at org.partiql.lang.CompilerPipelineImpl.compile(CompilerPipeline.kt:258)
	at org.partiql.pipeline.AbstractPipeline$PipelineStandard.compile(AbstractPipeline.kt:94)
	at org.partiql.shell.Shell$run$$inlined$use$lambda$3.invoke(Shell.kt:214)
	at org.partiql.shell.Shell$run$$inlined$use$lambda$3.invoke(Shell.kt:81)
	at org.partiql.shell.Shell.executeAndPrint(Shell.kt:224)
	at org.partiql.shell.Shell.run(Shell.kt:210)
	at org.partiql.shell.Shell.start(Shell.kt:108)
	at org.partiql.cli.Main.runShell(main.kt:252)
	at org.partiql.cli.Main.main(main.kt:203)
Caused by: org.antlr.v4.runtime.InputMismatchException
	at org.antlr.v4.runtime.DefaultErrorStrategy.sync(DefaultErrorStrategy.java:270)
	at org.partiql.lang.syntax.antlr.PartiQLParser.statement(PartiQLParser.java:543)
	... 13 more

ERROR!
OK!
```

## To Reproduce
Steps to reproduce the behavior:
1. Open CLI/REPL -- `./cli/shell.sh`
2. Type syntactically valid query and hit enter once
3. Type `!!` and hit enter again
4. Error

## Expected Behavior
Expect AST of syntactically valid query to be output.

## Additional Context
- Java version: 11
- PartiQL version: 0.8.2-SNAPSHOT-6550a75d

--- a/cli/src/main/kotlin/org/partiql/shell/Shell.kt
+++ b/cli/src/main/kotlin/org/partiql/shell/Shell.kt
@@ -36,6 +36,7 @@ import org.partiql.lang.eval.ExprValue
 import org.partiql.lang.eval.ExprValueFactory
 import org.partiql.lang.eval.PartiQLResult
 import org.partiql.lang.eval.delegate
+import org.partiql.lang.syntax.PartiQLParserBuilder
 import org.partiql.lang.util.ConfigurableExprValueFormatter
 import org.partiql.lang.util.ExprValueFormatter
 import org.partiql.pipeline.AbstractPipeline
@@ -152,6 +153,12 @@ internal class Shell(
                 return
             }
 
+            // Pretty print AST
+            if (line.endsWith("\n!!")) {
+                printAST(line.removeSuffix("!!"))
+                continue
+            }
+
             if (line.isBlank()) {
                 out.success("OK!")
                 continue
@@ -262,6 +269,17 @@ internal class Shell(
         val properties = Properties()
         properties.load(this.javaClass.getResourceAsStream("/partiql.properties"))
         return "${properties.getProperty("version")}-${properties.getProperty("commit")}"
+    }
+
+    private fun printAST(query: String) {
+        if (query.isNotBlank()) {
+            val parser = PartiQLParserBuilder.standard().build()
+            val ast = parser.parseAstStatement(query)
+            val explain = PartiQLResult.Explain.Domain(value = ast, format = null)
+            val output = ExplainFormatter.format(explain)
+            out.println(output)
+            out.flush()
+        }
     }
 
     /**

